lib/src/signed_video_h26x_auth.c

```diff
@@ -964,3 +967,42 @@ signed_video_add_nalu_and_authenticate(signed_video_t *self,
 
   return svi_rc_to_signed_video_rc(status);
 }
+
+
+SignedVideoReturnCode
+signed_video_set_public_key(signed_video_t *self, sign_algo_t algo, const char *public_key, size_t public_key_size)
+{
+  if (!self || !public_key || public_key_size == 0) return SV_INVALID_PARAMETER;
+  if (self->signature_info->public_key) return SV_NOT_SUPPORTED;
+
+  // Return SV_NOT_SUPPORTED if not called from start of stream
+  if (self->nalus_in_sei == true) return SV_NOT_SUPPORTED;
+
+  uint8_t *new_public_key = NULL;
+  svi_rc status = SVI_UNKNOWN;
+  SVI_TRY()
+    SVI_THROW_IF_WITH_MSG(
+        algo < 0 || algo >= SIGN_ALGO_NUM, SVI_NOT_SUPPORTED, "Algo is not supported");
+    // Make sure we have allocated enough memory
+    if (self->signature_info->public_key_size != public_key_size) {
+      new_public_key = realloc(self->signature_info->public_key, public_key_size);
+      SVI_THROW_IF(!new_public_key, SVI_MEMORY);
+      self->signature_info->public_key = new_public_key;
+    }
+    memcpy(self->signature_info->public_key, public_key, public_key_size);
+
+    self->signature_info->algo = algo;
+    self->signature_info->public_key_size = public_key_size;
+    self->has_public_key = true;
+
+  SVI_CATCH()
+  {
+    // Remove all key information if we fail.
+    free(self->signature_info->public_key);
+    self->signature_info->public_key = NULL;
+    self->signature_info->public_key_size = 0;
```

Comment:
These are not necessary, since the only case that can fail is when public_key is already NULL.